### DISCUSSION TITLE
It's possible to have the same file associated with multiple models?

### DISCUSSION THREAD

**Discussion by fabiomlferreira** [spatie/laravel-medialibrary#2021] [Q&A]
In some cases I will need to have the same file associated with more that one model, for example I have a store with products, each product have one file associated when I sell that product I associate the current file to that order. At that point the same file is associated with the product and the order. In the future I can change the product file but keep for the old order the same file.

In this case scenario I need to have the same file associated with more than one model this is possible?

It's also useful for example in a blog have the same image associated in more that one post.

**Comment by freekmurze** [comment]
This is not possible at the moment.

**Reply by fabiomlferreira** [reply]
@freekmurze This was a excelent new feature. As this time this limitation of a file be associated with a single model is the only real problem with this module. If we want to create a news section and reuse the same image over and over again we need to create table for images, then adde the image to each entry on that table and then in the post associate the image_id from that table. 

If we can add multiple models to a single file it was perfect to create a filemanager without the need of create a new table on our project.
